### ISSUE TITLE
Apple Silicon bundle of Frescobaldi 3.3 does not run (“file damaged”)

### ISSUE THREAD

**Issue by TonioMus** [frescobaldi/frescobaldi#1576]
<Please describe the issue or feature request.
Provide as much information as possible.>

Frescobaldi: 3.3.0
Extension API: 0.9.0
Python: 3.10.10
python-ly: 0.9.7
Qt: 5.15.3
PyQt: 5.15.7
sip: 6.6.2
qpageview: 0.6.2
poppler: 22.9.0
python-poppler-qt: 21.3.0
OS: macOS-13.2.1-x86_64-i386-64bit
installation kind: standalone .app bundle

**Comment by TonioMus** [comment]
Using a MacMini M1, I followed the link proposed by SourceForge to download the new Frescobaldi-3.3.0-arm64.dmg file.
Download ok, dmg decoded and mounts correctly ; App file copied in the App folder, but when trying to launch it : "Impossible to run, file damaged"
Tried Three times, after reboot the Mac : same problem.
The i86 version installed perfectly and seems to work well.
Regards
Antoine

**Comment by TonioMus** [comment]
Oops ! Jeanas, I was writing and didn't see your answer

**Comment by TonioMus** [comment]
Clear

**Comment by TonioMus** [comment]
so, as I can survive with the new i86 version, I can let you have a look to the silicon one problem quietly !

**Comment by jacques-menu** [comment]
Hello TonioMus,

I built the 3.3.0 for arm64 and use it here without any problem. 
How can I send it to you to try it, until the issue is solved? 

JM

**Comment by TonioMus** [comment]
Oops, excuse me, I was on another screen for another job and no "big" to alert me
Dropbox looks good for me, or propose me something else

**Comment by TonioMus** [comment]
"no bip" I meant

**Comment by jacques-menu** [comment]
Pièce jointe disponible jusqu’au 1 mai 2023
Hello TonioMus,

Here is the one that runs fine here, can you check it on your machine, just to be sure it works elsewhere? Thanks!

Cliquer pour télécharger
 <https://www.icloud.com/attachment/?u=https%3A%2F%2Fcvws.icloud-content.com%2FB%2FAQaSLtTFrJT9ZvyHnZM6gNWCTSxIAScXM3TQA_wC3vBW2w6LW_jHy9lc%2F%24%7Bf%7D%3Fo%3DAjmmbGHz7K3HXbsDIc599kHMaerMA9vo5Pf8ghdIe2Xb%26v%3D1%26x%3D3%26a%3DCAogXnUTnpu0v4paeKcE7oCI6WDU4gwIEunOb4LTKMrKT-kSdhCSi87s8zAYkpvJwP0wIgEAKgkC6AMA_0j1jQVSBIJNLEhaBMfL2VxqJWQ9OXgv31uzjL8Bxfx60JCoJuuawPJNB2VSAkFFR8Hy-Nqlu8tyJfazgixs7tDUSvUY4XHBmmkyuzBTzlOeJSkBdUiqiLRB5SZqxQs%26e%3D1682957290%26fl%3D%26r%3D691DA9CA-862B-4B99-8905-B603E9F73C33-1%26k%3D%24%7Buk%7D%26ckc%3Dcom.apple.largeattachment%26ckz%3D219B9D92-907C-4C52-8F39-AED33116FB66%26p%3D53%26s%3DPXervIf376yiKBp60pZUtXWxyL0&uk=hiWGcZ_NlKY7QN47ZNRunQ&f=Frescobaldi.app.zip&sz=68116786>Frescobaldi.app.zip
179,2 Mo

**Comment by TonioMus** [comment]
iCloud was a natural and better solution (-; !
Well, got the file and copied in the Apps folder, and… same error alert window )-: "Frescobaldi.app » est endommagé et ne peut pas être ouvert"

**Comment by jacques-menu** [comment]
Did users report about 3.3.0 running fine on their machine?

JM

**Comment by TonioMus** [comment]
I wrote earlier that i86 3.3.0 version is running fine on my Mac mini M1

I just experienced a problem with the arm 
file "

**Comment by TonioMus** [comment]
I noticed that, when the i86 version was blocked at the first launch time by the common "Apple security lock", I could unlock as usual in such case, 
for the arm file, the process didn't go so far : it was stopped at the first attempt of opening of the file.

**Comment by TonioMus** [comment]
Ok, Jeanas, I may be the first human being to have tried !

Thanks for your help and see you later, for I have to concentrate to another job.
Read you soon
Regards
Antoine (music, networking and electronics)

**Comment by ContraIndiKate** [comment]
> Using a MacMini M1, I followed the link proposed by SourceForge to download the new Frescobaldi-3.3.0-arm64.dmg file. Download ok, dmg decoded and mounts correctly ; App file copied in the App folder, but when trying to launch it : "Impossible to run, file damaged" Tried Three times, after reboot the Mac : same problem. The i86 version installed perfectly and seems to work well. Regards Antoine

===
I am experiencing exactly the same issue on a MacBook Pro M1 running Monterey 12.6.3.

**Comment by TonioMus** [comment]
Hello ContralndiKate.
The OS running on may MacMini M1 is Ventura 13.2.1
8 Go mem. and 500 Go SSD

**Comment by ContraIndiKate** [comment]
Hi TonioMus, that's good to know, thanks. I was wondering if I needed to update to get this to work.

**Comment by TonioMus** [comment]
Super ContalndiKate
That’s sure the reason why they recommend us always to give the max infos we can (-; !

**Comment by dliessi** [comment]
I believe that this is due to the fact that our app bundles are not signed. See for example https://github.com/archimatetool/archi/issues/555.

There is a workaround, but (when it works) it completely avoids Gatekeeper's checks, which is a security risk for the user. I completely trust @jacques-menu, who built the arm64 application, but I appreciate that some users may not trust us enough.

The workaround consists in removing the quarantine flag from the app bundle by opening a Terminal and running the following command:
```
sudo xattr -r -d com.apple.quarantine /Applications/Frescobaldi.app
```

(I'm assuming that the user copied the application bundle to the `/Applications` directory. If that is not the case the actual path to the app needs to be specified. The easiest way is the following: write in the Terminal `sudo xattr -r -d com.apple.quarantine ` without pressing enter at the end, final space included; drag and drop the app bundle on the Terminal window, making the full path to the app should appear at the end of the command; then press enter.)

@TonioMus, @ContraIndiKate: if you trust us enough, could you please try the workaround and report back?

The only way to solve the problem requires signing the application bundles in the Apple-approved way. This includes paying for an Apple developer account (99$/year). I'm not going to do it, so for now if the workaround indeed works I'm going to add a note to the release page.

If anyone has a suitable Apple account and volunteers to sign our application bundles, they're more than welcome.

**Comment by TonioMus** [comment]
Thanks a lot dliessi.
Of course, I'll try what you propose within 5 mn.
Remembre although what I have written earlier :
"I noticed that, when the i86 version was blocked at the first launch time by the well known "Apple security lock", I could unlock as usual in such case*,
For the arm file, the process didn't go so far : it was stopped at the first attempt of opening of the file."
* I meant by going in the "system tuning" and check to allow the usigned  bundle to be used.

So do I suspect there may exist an other case…

But as promised, let's go and check

**Comment by dliessi** [comment]
By reading more into the linked issue, there might be a way to at least avoid the "damaged file" message and get the "unidentified developer" one, which can overridden by the user. The "damaged file" message may be caused by a partial signature of the app bundle and the workaround removes all code signatures.

@TonioMus, @ContraIndiKate: you might try the following, possibly copying the app bundle from the DMG again, if you tried the other workaround in the meantime):
```
codesign --remove-signature /Applications/Frescobaldi.app
```
The same comment about the path to the app as in my previous message applies also here.

After running this command, when you try to open Frescobaldi you should get a message like the one shown [here](https://github.com/archimatetool/archi/issues/555#issuecomment-751356113). After clicking OK to close the warning, you can right-click on the application bundle and there should be an option "Open anyway".

If this work, we can add the signature removal step in the build process.

**Comment by TonioMus** [comment]
Well, I did it ! (simply with "sudo xattr -r -d com.apple.quarantine /Applications/Frescobaldi.app" as you proposed.
It took me quite a long time because of a sudo password mess (not recognized when logged as a "normal" user ; phew…)

**Comment by TonioMus** [comment]
[test Fresco 3-3 TonioMus.pdf](https://github.com/frescobaldi/frescobaldi/files/11130594/test.Fresco.3-3.TonioMus.pdf)

**Comment by dliessi** [comment]
@TonioMus Good news, thank you!

Would you mind trying also the other workaround (on another fresh copy of the app bundle)?

**Comment by TonioMus** [comment]
I was going the other way (alway long because of the need to switch for normal to admin, and so on…

**Comment by TonioMus** [comment]
oops –> same message : "fichier endommagé" !

The result of the command in terminal mode was not as good as I could have guessed : see the attached file
[Result.pdf](https://github.com/frescobaldi/frescobaldi/files/11130626/Result.pdf)

**Comment by TonioMus** [comment]
I meant I was surprised by the result…

[Result meth 2.txt](https://github.com/frescobaldi/frescobaldi/files/11130652/Result.meth.2.txt)

**Comment by TonioMus** [comment]
Hello everybody !
Have a perfect Sunday.

Fisrt I hope @:ContraIndiKate succeeded in running this new version, either the i86, or, better the arm one (as I did again this morning).

Second, reading the following point (from @:dliessi)
_The only way to solve the problem requires signing the application bundles in the Apple-approved way. This includes paying for an Apple developer account (99$/year). I'm not going to do it, so for now …_

I think that 
on one hand a lot of Mac users appreciate Apple have been proposing and managing different ways of securing what may happen with apps and on the web…
on the other hand the same people appreciate "free softwares" and understand that the developers involved in creating and maintenaient such apps don't want to pay, even for honest people that would additionally wish to be secured, but for free, again !
So, my question is : as there exist technical means to share a unique Apple account for a group of developers / one app (my son Roland did so), would it be really impossible to gather the yearly 99$  by the way of collecting the contributions of a reasonable number of users/volunteers ?

Read you soon
(and luck and courage to solve the issue !)
Antoine

**Comment by ContraIndiKate** [comment]
Thanks @dliessi and @TonioMus for your work on this! I'm on a crunchy deadline so I reverted back to the last stable version to get going on my project again. I'll have some time in a few weeks so I'll check back then to see if I can help out with this.

**Comment by TonioMus** [comment]
Okay @:ContralndiKate, this is wise.
I keep the 3.3.0 ARM version – with the "… - quarantine" trick instead, just because I'm so happy to have an installable version for the M1, at last !!
Since I have bought the Mac mini M1, I have been using BBEdit as editor and was launching Lilipond from the Terminal window !
It was not so easy, but efficient (-: !

A lot of courage for your project.
A

**Comment by dliessi** [comment]
@TonioMus, @ContraIndiKate, @dmingram66: can you please try [this app bundle (link to iCloud)](https://www.icloud.com/attachment/?u=https%3A%2F%2Fcvws.icloud-content.com%2FB%2FAWrlN2zI2Uk-7SsH1BhzStcoNXFFAbywEHkPSc76pKEFoTLdLTMh6xU2%2F%24%7Bf%7D%3Fo%3DAux6BxarBl1t8XHEr77HD3LPUrDVbdRC2hvwQNVIJiL8%26v%3D1%26x%3D3%26a%3DCAogGznq2VDQ1J24TOPDXDKp21yMGNJJ4RN-eBtgu57sSMkSdhCC_7eF9DAYgo-z2f0wIgEAKgkC6AMA_1fAqsBSBCg1cUVaBCHrFTZqJS6m-yx9pLmmHwAw_Mubmc3QjXqgmR7hEAaGFUvy_lbC2wRipO1yJf0UT1xvmBeuxOhViQ8nNvISt8yBuKaQEPs2gvDEq3ZdPOMGTIM%26e%3D1683009357%26fl%3D%26r%3D13F5C9F4-E5C8-44A9-8598-212C88950F42-1%26k%3D%24%7Buk%7D%26ckc%3Dcom.apple.largeattachment%26ckz%3D219B9D92-907C-4C52-8F39-AED33116FB66%26p%3D53%26s%3DE_LY_3boQNBGj7pfCj-QkwqOJVw&uk=Ed-xJjIKp5bFn1LBpLmtgw&f=Frescobaldi.app.zip&sz=67953582)?

We stripped the partial signature as described in https://github.com/frescobaldi/frescobaldi/issues/1576#issuecomment-1493117552. Trying to open it should tell you that Gatekeeper could not identify the developer, or that it could not check the app for malicious software, but it should not tell you that the app is damaged anymore. You should be able to right-click on the app and choose "Open anyway". Thank you.

**Comment by TonioMus** [comment]
Ok, thksI’ll try it just after dinner.AntoineLe 2 avr. 2023 à 19:46, Davide Liessi ***@***.***> a écrit :﻿
@TonioMus, @ContraIndiKate, @dmingram66: can you please try this app bundle (link to iCloud)?
We stripped the partial signature as described in #1576 (comment). Trying to open it should tell you that Gatekeeper could not identify the developer, or that it could not check the app for malicious software, but it should not tell you that the app is damaged anymore. You should be able to right-click on the app and choose "Open anyway". Thank you.

**Comment by TonioMus** [comment]
Hum…
Don't behave as expected (see attachements).
[2avril20h53.pdf](https://github.com/frescobaldi/frescobaldi/files/11133078/2avril20h53.pdf)
[2avril20h53 - right click.pdf](https://github.com/frescobaldi/frescobaldi/files/11133079/2avril20h53.-.right.click.pdf)

**Comment by TonioMus** [comment]
Further, even after having added the line "(open from) anywhere" ie "n'importe où" in French, the app cannot be opened.
[2avril21h18 from anywhere.pdf](https://github.com/frescobaldi/frescobaldi/files/11133114/2avril21h18.from.anywhere.pdf)

**Comment by dliessi** [comment]
@TonioMus: Just to be sure of the steps: do you get those messages when right-clicking on the app and selecting "ouvrir"?

**Comment by TonioMus** [comment]
Yes, as you demanded : "right-click"

**Comment by jacques-menu** [comment]
Ciao Davide,

Buying Apple Developer Program membership is meant for individuals and organisations. In the latter case, there is a verification process involved, which I doubt crowd funding would pass successfully.

I could buy that as an individual, but then I’d appear as Fresco’s developper, which wouldn’t be fair of course…

So what do you think?

JM

**Comment by TonioMus** [comment]
Hello all the dev. Team.

A great thanks for your work ; as a regular & happy user of Frescobaldi/Lilypond, and as a retired electronic and computing engineer as well, I appreciate a lot…

Of course, the good idea would be the « organization » way, not the individual one.
So, an official « organization shall exist ; is it a great problem ? I don’t know…
Anyway, it wouldn’t sound strange to me that a team like the one developing the program appears somewhere, for some (good) reasons as an « organization » ; further, I have no doubt concerning the result of a crowdfunding to feed the cash.

Regards
Antoine
-
Copy to Roland, a son of mine, expert in computing technologies
-

**Comment by jacques-menu** [comment]
Hello,

What I could do is to pay the Apple Developer Program membership for Wilbert, assuming he would agree.
That would be my contribution to this great and useful piece of software.

JM

**Comment by TonioMus** [comment]
I would like du contribute too !AntoineLe 5 avr. 2023 à 16:02, jacques-menu ***@***.***> a écrit :﻿
Hello,

What I could do is to pay the Apple Developer Program membership for Wilbert, assuming he would agree.
That would be my contribution to this great and useful piece of software.

JM

**Comment by dliessi** [comment]
There were some leftovers from the signature stripping (apparently `codesign --remove-signature` is a bit buggy). I cleaned up the app bundle.

@TonioMus: Can you test [this app bundle (wetransfer link)](https://we.tl/t-l4z4rIR1Ur)? Try also control-click, open (it may be not equivalent to right-click). Thank you.

**Comment by nodevnul** [comment]
> There were some leftovers from the signature stripping (apparently `codesign --remove-signature` is a bit buggy).>

The command is not buggy. It works as intended. Explanations see here:

https://eclecticlight.co/2020/11/03/code-signatures-3-fixing-signature-problems/

**Comment by nodevnul** [comment]
Codesigning, Signatures, Apple Developer Account

Although software intended to run on Apple Silicon ARM hardware is required to be code signed, an Apple Developer Account is not needed to achieve the requirements:

https://eclecticlight.co/2020/08/22/apple-silicon-macs-will-require-signed-code/

Citation:
„Two important points need to be reiterated: you don’t need a developer signature, and can use an ad-hoc one, even one generated on the fly during the build process, and this only applies to ARM native executables.”

The method is outlined here:

https://eclecticlight.co/2019/01/17/code-signing-for-the-concerned-3-signing-an-app/

Some more info here:

https://eclecticlight.co/2019/01/15/code-signing-for-the-concerned-1-why/
https://lapcatsoftware.com/articles/hardened-runtime-sandboxing.html

About signatures here:

https://eclecticlight.co/2020/11/03/code-signatures-3-fixing-signature-problems/


It appears that guessing is not appropriate to handle issues. The project albeit needs to discuss if ad hoc signing is feasable in the context of trust and compliance.

**Comment by dliessi** [comment]
@nodevnul Thanks for the information. Ad hoc signing does not mean anything from a security point of view, indeed, but at least it should allow users to run the application. Let's give it a try.

@jacques-menu Can you please do
```
codesign --force --deep -s - Frescobaldi.app
```
as suggested in https://eclecticlight.co/2019/01/17/code-signing-for-the-concerned-3-signing-an-app/ and post here a link to the resulting app?
I tried on my machine but got an error about a missing `codesign_allocate` tool, which in fact is where it should be.

**Comment by jacques-menu** [comment]
Pièce jointe disponible jusqu’au 6 mai 2023
Hello Davide,

***@***.***:/Volumes/JMI_VOLUME/JMI_Documents > codesign --force --deep -s - Frescobaldi.app
Frescobaldi.app: replacing existing signature

Here is the signed app:

Cliquer pour télécharger
 <https://www.icloud.com/attachment/?u=https%3A%2F%2Fcvws.icloud-content.com%2FB%2FAbO7GfIgfmUMFgkx9MhUsQzb_B9rAT7LpOJ6P_pfGWyl7yE9se_-uEX7%2F%24%7Bf%7D%3Fo%3DAkyxzyZFn1-lXRBmGFsJMZjD-YnD2J_8L9uPX0Tiu3L4%26v%3D1%26x%3D3%26a%3DCAogXcceDU2dWj71KayrxEsU-8YyBTf498ngKCoaiswyNwYSdhCTruSq9TAYk77f_v4wIgEAKgkC6AMA_3LYkhFSBNv8H2taBP64RftqJedM_Tlid0u6LmhsAgymCEt1k6BdzQx3x3chQl1yIYwAakA_LVdyJR0nlVLYIYeou_Mq-bCc6riXHdv8q_bDz1jJ83gzf6GbZOY0o10%26e%3D1683356114%26fl%3D%26r%3DB8754FD6-1CBE-4900-915C-22DF884B5431-1%26k%3D%24%7Buk%7D%26ckc%3Dcom.apple.largeattachment%26ckz%3D219B9D92-907C-4C52-8F39-AED33116FB66%26p%3D53%26s%3DkGZBH-FXdUUZWh-D0lBiB-NhbHA&uk=34iboHcUuDb4ts0-uQV9tw&f=Frescobaldi.zip&sz=66903510>Frescobaldi.zip
66,9 Mo

JM

**Comment by TonioMus** [comment]
Hello JM
Here is the result )-: !
Courage !
Antoine
-
"Unable to open
"Frescobaldi.app" because Apple cannot verify that it is free of malware.
This software needs to be updated. Contact the developer for a better knowledge.
Safari downloaded this file today at 09:31."
[Test FrescobARM ao apr6 9-32.pdf](https://github.com/frescobaldi/frescobaldi/files/11167548/Test.FrescobARM.ao.apr6.9-32.pdf)

**Comment by dliessi** [comment]
@jacques-menu Thank you!

@TonioMus That's a step forward: ad hoc signatures are still blocked by Gatekeeper, but at least the application is now correctly signed. If you now control-click on it and select "Open" you should get a similar message, but with also an "Open" or "Open anyway"  button and you should be able to run the application. Can you please try?

**Comment by dliessi** [comment]
I had a report of success in opening the new correctly signed application. I'm going to prepare and upload the new DMG right away. Thanks to everyone for the help!

**Comment by TonioMus** [comment]
Okay
Done and working
thés a lot.
Antoine

**Comment by TonioMus** [comment]
Oops !
Yes, it opens (after a first opening with ctrl-click and choosing “open anyway”) but
The app main window doesn’t appear every time and when it does, impossible to act (click for any action, edit, …) inside it !

**Comment by dliessi** [comment]
@TonioMus Known issue #428: click on the app icon on the dock after the splashscreen disappears and the main window will appear.

**Comment by dliessi** [comment]
Sorry, I didn't read the last half sentence... If the window appears _without_ clicking on the dock icon and is not responsive, maybe clicking on the dock icon makes it responsive. Can you please check next time it happens? If clicking on the dock icon solves the problem, please add a comment to #428; otherwise please open a new issue for that problem.

**Comment by dliessi** [comment]
I'm closing this issue, as the original problem (damaged app) is solved. I'll open a new issue for the wish of signing with an Apple Developer certificate, but I consider this low priority until we can build a fully functional application bundle (see #1244).

**Comment by nodevnul** [comment]
@ALL

@dliessi Thanks for mentioning

Thanks for the tremendous efforts. I can report

• macOS Ventura 13.3 (22E252)
• MBP M1Max
• Downloaded 3.3.0 Arm64

• 1st Control-Right-Click -> Open: OS Message „Cannot be opened” (as expected)
• 2nd Control-Right-Click -> Open: Frescobaldi launches fast - very fast…

• After Splashscreen App is in Focus with the default empty project window - no need to click on App-Icon in Dock to bring Frescobaldi in focus (at least for this launch iteration)

Right now I am missing an Lilypond installation because I couldn't manage to monitor the status of Apple Silicon ARM64 binaries of the Lilypond package. I also am not up to date if the Lilypond Intel-Binaries can be used from within Frescobaldi ARM64.

**Comment by jacques-menu** [comment]
Hello nodevnull,

I run the Fresco arm64 v3.3.0 version I ‘ve built thanks to Davide’s help together with LilyPond from MacPorts. Sometimes Fresco freezes or has this menu bar display problem, otherwise the combination works fine.

JM

**Comment by jacques-menu** [comment]
PS> I volunteer to build LilyPond on MacOS with Apple Silicon, but I’ll need help the kind Davide provided for Fresco.

JM

**Comment by nodevnul** [comment]
Ok.

I have installed the 2.25.3 Intel binaries from the Lilypond website. Unfortunately the user has to go through the Control-Right-Click -> Open for all executables that could possibly be launched before pointing Frexcobaldi to use them for rendering. Otherwise rendering is not possible because for each binary macOS will show the verification error message.

That's a bummer.

**Comment by TonioMus** [comment]
> > I also am not up to date if the Lilypond Intel-Binaries can be used from within Frescobaldi ARM64.
> 
> They can. You could also install LilyPond from Homebrew or MacPorts, both of which provide native arm64 binaries.

Hello !
Yes, I can confirm : I did it two days ago (ie install & run intel's V3.3.0 on my Mac mini M1 / OS 13.2.1) and Lilypond* 2.22.1 that was installed far before with Homebrew.

* Since November 2022, I have been using this Lilypond 2.22.1 in "command line" mode (along with BBEDIT as an editor !!)
